lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenProjectCache.java

```diff
@@ -230,12 +273,14 @@ public void addProjectParsedListener(Consumer<MavenProject> listener) {
 		this.projectParsedListeners.add(listener);
 	}
 
+	@Deprecated
 	public DefaultRepositorySystemSession getRepositorySystemSession() {
 		try {
 			initializeMavenBuildState();
 		} catch (Exception e) {
 			e.printStackTrace();
 		}
+		// TODO: How does this initialize the repository system session?
```

Comment:
@mickaelistria am I missing something here? Or is it ok to remove/deprecate this method? I'm now passing the repositorySystemSession from MavenPlugin

Reply:
Yes, it seems good to remove ti.

Reply:
Actually, you may want to remove initializeMavenBuildState by the way and make the projectBuilder and the DISABLE_GLOBAL_MODEL_CACHE_SYSTEM_PROPERTY system property set in the constructor.

Reply:
Good call, I'll try that

Reply:
I ended up moving initializeMavenBuildState to the constructor.